kuma/users/templatetags/jinja_helpers.py

```diff
@@ -12,15 +12,22 @@
 from kuma.core.templatetags.jinja_helpers import datetimeformat
 from kuma.core.urlresolvers import reverse
 
-from ..jobs import UserGravatarURLJob
-
 
 @library.global_function
-def gravatar_url(email, secure=True, size=220, rating='pg',
-                 default=settings.DEFAULT_AVATAR):
-    job = UserGravatarURLJob()
-    return job.get(email, secure=secure, size=size,
-                   rating=rating, default=default)
+def get_avatar_url(user, provider='github'):
+    """
+    Get the user's avatar URL for the specified provider. Returns None if the
+    user is not logged-in, or the default avatar if the user is logged-in but
+    has no avatar for the specified provider.
+    """
+    if user and user.is_authenticated:
```

Comment:
This will break the syndication feed. 

All places except one already do a check that `user` is a thing and `.is_authenticated`. E.g. `login.html` does:
```django
{% if user.is_authenticated %}
        <a class="user-url" href="{{ user.get_absolute_url() }}">
            <img src="{{ get_avatar_url(user) }}" 
...
```
Same with the `whoami` view:
```python
user = request.user
    if user.is_authenticated:
        data = {
            'username': user.username,
            'timezone': user.timezone,
            'is_authenticated': True,
            'is_staff': user.is_staff,
            'is_superuser': user.is_superuser,
            'is_beta_tester': user.is_beta_tester,
            'avatar_url': get_avatar_url(user),
```

However, one place where it doesn't is the `DocumentJSONFeedGenerator.write` method:

```python
if revision.creator:
    item_out['author_avatar'] = get_avatar_url(revision.creator)
```
In this case, the `user` isn't from a `request` but rather from a regular `wiki.Document.current_revision` lookup.

I think we should simply delete this if-statement and it should work in both scenarios. The feed would get avatar URLs.

Reply:
Good point. Although the `if user and user.is_authenticated:` is probably harmless (since it's really meant to filter-out anonymous users), I can't think of any case where we'd call `get_avatar_url` without testing for the user anyway (since we'd have to handle the `None` returned otherwise). I'll remove that check. Thanks @peterbe!